sale_cancel_reason/wizard/cancel_reason.py

```diff
@@ -0,0 +1,27 @@
+# © 2013 Guewen Baconnier, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import models, fields, api
+
+
+class SaleOrderCancel(models.TransientModel):
+
+    """ Ask a reason for the sale order cancellation."""
+    _name = 'sale.order.cancel'
+    _description = __doc__
+
+    reason_id = fields.Many2one(
+        'sale.order.cancel.reason',
+        string='Reason',
+        required=True)
+
+    def confirm_cancel(self):
+        self.ensure_one()
+        act_close = {'type': 'ir.actions.act_window_close'}
+        sale_ids = self._context.get('active_ids')
+        if sale_ids is None:
+            return act_close
+        assert len(sale_ids) == 1, "Only 1 sale ID expected"
```

Comment:
Please use UserError. We are in an Odoo context, so using its exceptions is better.